### PR TITLE
New ol.proj features

### PR DESCRIPTION
**Pull request by bill-chadwick**
Add support for custom getPointResolution,
setting the 'global' boolean
and creating proj4 projections with options.

### CODE REVIEW

**Comment by ahocevar** [comment]
As discussed in #3285, I'm not so sure about `addProj4Projection()` any more. Also see #3362 for an alternative to it.

Other than that, the content of this pull request makes great sense to me.

**Comment by bill-chadwick** [comment]
addProj4Projection removed.

File: src/ol/proj/proj.js [hunk 1]
```diff
@@ -285,6 +315,26 @@ ol.proj.Projection.prototype.getPointResolution = function(resolution, point) {
...
+ * the point resolution by transforming the 'point' pixel to EPSG:4326,
+ * measuring its width and height on the normal sphere,
+ * and taking the average of the width and height.
+ * An alternative implementation may be given when constructing a
+ * projection (or creating one from proj4). For many local projections,
+ * such a custom function will return the resolution unchanged.
+ * @param {number} resolution Resolution in projection units.
+ * @param {ol.Coordinate} point Point.
+ * @return {number} Point resolution in projection units.
+ * @api stable
```
Comment (ahocevar) [inline src/ol/proj/proj.js]: not stable please.

**Comment by ahocevar** [comment]
Looks like you got a commit with comment "squash" when you tried to squash two commits into one. If you can fix this, I'm going to merge as soon as Travis gives the ok.

**Comment by bill-chadwick** [comment]
Tidied up now I hope. Done for tonight.

**Comment by ahocevar** [comment]
Great work, thanks again!

**Pull request merged**